src/js/tools/type.js

```diff
@@ -70,80 +84,66 @@ define(function (require, exports, module) {
+                    return;
+                }
 
-                    if (event.alignAssigned) {
-                        this.flux.actions.type.updateAlignment(currentDocument, layers, event.align);
-                    } else {
-                        this.flux.actions.type.updateAlignment(currentDocument, layers, null);
-                    }
+                var layerID = event.layerID,
+                    currentLayer = document.layers.byID(layerID);
 
-                    // set leading to null if autoleading is true
-                    if (event.autoLeadingAssigned && event.autoLeading) {
-                        this.flux.actions.type.updateLeading(currentDocument, layers, null);
-                    } else {
-                        if (event.leadingAssigned) {
-                            this.flux.actions.type.updateLeading(currentDocument, layers, event.leading);
-                        } else {
-                            this.flux.actions.type.updateLeading(currentDocument, layers, undefined);
-                        }
-                    }
+                if (currentLayer) {
+                    log.warn("Unexpected createTextLayer event for layer " + layerID);
+                    this.flux.actions.layers.resetLayers(document, currentLayer);
+                } else {
+                    _layersReplaced = false;
+                    this.flux.actions.layers.addLayers(document, layerID, true)
+                        .then(function (layersReplaced) {
+                            _layersReplaced = layersReplaced;
+                        });
                 }
             }.bind(this);
+            
+            descriptor.addListener("createTextLayer", _layerCreatedHandler);
 
-            descriptor.addListener("updateTextProperties", _typeChangedHandler);
-
-            _layerCreatedHandler = function (event) {
+            _layerDeletedHandler = function (event) {
                 var documentStore = this.flux.store("application"),
-                    currentDocument = documentStore.getCurrentDocument(),
-                    layerID = event.layerID,
-                    currentLayer = currentDocument.layers.byID(layerID);
+                    document = documentStore.getCurrentDocument();
 
-                if (currentLayer) {
-                    log.warn("Unexpected createTextLayer event for layer " + layerID);
-                    this.flux.actions.layers.resetLayers(currentDocument, currentLayer);
+                if (!document) {
+                    log.error("Unexpected deleteTextLayer event: no active document");
+                    return;
+                }
+
+                var layerID = event.layerID,
+                    layer = document.layers.byID(layerID);
+
+                if (layer) {
+                    if (_layersReplaced) {
+                        // See comment above at the _layersReplaced declaration
+                        this.flux.actions.documents.updateDocument();
```

Comment:
Actually I think I'm against adding that extra line, since it will make the supposed state machine more obscure from our end, and there's no way to defend against all possible PS bugs.